src/components/common/quantityInterval/QuantityInterval.tsx

```diff
@@ -1,68 +1,33 @@
-import React, { useState, useEffect, useCallback } from "react";
-import { FormControl } from "@mui/material";
+import React, { useState } from "react";
 import "twin.macro";
 import "styled-components/macro";
-import * as ucum from "@lhncbc/ucum-lhc";
 import Quantity from "../quantity/Quantity";
+import { CQL } from "cqm-models";
 
 interface QuantityIntervalProps {
   label: string;
-  lowQuantity: number;
-  handleLowQuantityChange: Function;
-  lowQuantityUnit: any;
-  handleLowQuantityUnitChange: Function;
-  highQuantity: number;
-  handleHighQuantityChange: Function;
-  highQuantityUnit: any;
-  handleHighQuantityUnitChange: Function;
+  lowQuantity: CQL.Quantity;
+  highQuantity: CQL.Quantity;
   canEdit: boolean;
 }
 
 const QuantityInterval = ({
   label,
   lowQuantity,
-  handleLowQuantityChange,
-  lowQuantityUnit,
-  handleLowQuantityUnitChange,
   highQuantity,
-  handleHighQuantityChange,
-  highQuantityUnit,
-  handleHighQuantityUnitChange,
   canEdit,
 }: QuantityIntervalProps) => {
-  const [ucumOptions, setUcumOptions] = useState([]);
-  const [ucumUnits, setUcumUnits] = useState([]);
-
-  const buildUcumUnits = useCallback(() => {
-    const options = [];
-
-    for (const [key, value] of Object.entries(ucumUnits)) {
-      const current = value;
-      const { csCode_, guidance_, name_ } = current;
-      const option = {
-        code: csCode_,
-        guidance: guidance_,
-        name: name_,
-        system: "https://clinicaltables.nlm.nih.gov/",
-      };
-      options.push(option);
-    }
-    setUcumOptions(options);
-  }, [ucumUnits, setUcumOptions]);
-
-  useEffect(() => {
-    if (ucumUnits) {
-      buildUcumUnits();
-    }
-  }, [ucumUnits, buildUcumUnits]);
-
-  useEffect(() => {
-    if (!ucumUnits.length) {
-      ucum.UcumLhcUtils.getInstance();
-      const unitCodes = ucum.UnitTables.getInstance().unitCodes_;
-      setUcumUnits(unitCodes);
-    }
-  }, [ucum, ucumUnits]);
+  const [currentLowQuantity, setCurrentLowQuantity] =
+    useState<CQL.Quantity>(lowQuantity);
+  const [currentHighQuantity, setCurrentHighQuantity] =
```

Comment:
same as others, these internal state variables may not be needed as value should come from parent, and updated value is passed to parent via `onChange`